joblib/parallel.py

```diff
@@ -1276,31 +1273,30 @@ def _get_outputs(self, iterator, pre_dispatch):
             # the user if necessary.
             self._exception = True
 
-            # In CPython, the gc and the error happen in the main thread.
-            # However, this is not the case for other interpreters such as in
-            # PyPy, where this block can be executed in arbitrary threads. This
-            # can lead to hang when a process tries to join itself. As a
+            # In some interpreters such as PyPy or potentially CPython without
+            # GIL, this error might happen not in the main thread. This
+            # can lead to hang when a thread tries to join itself. As a
             # workaround, we detach the execution of the aborting code to a
             # dedicated thread. We then need to make sure the rest of the
             # function does not call `_terminate_and_reset` in finally.
-            if IS_PYPY and current_thread_id != threading.get_ident():
-                pypy_detach_generator_exit = True
+            if current_thread_id != threading.get_ident():
+                detach_generator_exit = True
                 _parallel = self
 
-                class _PypyGeneratorExitThread(threading.Thread):
+                class _GeneratorExitThread(threading.Thread):
                     def run(self):
                         _parallel._abort()
                         if _parallel.return_generator:
                             _parallel._warn_exit_early()
                         _parallel._terminate_and_reset()
 
-                _PypyGeneratorExitThread(
-                    name="PypyGeneratorExitThread"
+                _GeneratorExitThread(
+                    name="GeneratorExitThread"
                 ).start()
                 return
 
-            # Otherwise, we are in CPython or in PyPy in the main thread and
-            # we can safely abort the execution and warn the user.
+            # Otherwise, we are in the main thread and we can safely abort
+            # the execution and warn the user.
```

Comment:
```suggestion
            # Otherwise, we are in the thread that started the dispatch: we can
            # safely abort the execution and warn the user.
```